### ISSUE TITLE
shortcodes in included document

### ISSUE THREAD

**Issue by RobinD42** [getnikola/nikola#3206]
### Environment

**Python Version:** 3.6

**Nikola Version:** 8.0.1

**Operating System:** macOS

### Description:

I have a document that includes another using ReST's `.. include::` directive. In the included document, also a ReST doc, I use a shortcode like  `{{% name arg1 arg2 %}}` however it gets rendered as that exact text rather than invoking the shortcode. If I use the included document by itself without including it in another then the shortcode is rendered correctly. Expected behavior is that the shortcodes would be rendered in either case. Is there a workaround to get the shortcode rendered correctly when including the document?

**Comment by ralsina** [comment]
Sadly, because of how shortcodes are implemented, this is not very easy.

Basically, this is how it works now:

1. Get the post source
2. Replace the shortcodes with placeholders
3. Process with docutils or whatever compiler
4. Replace the placeholders with the HTML generated by the shortcodes

In your case, you are adding new shortcodes in step 3, which means step 4 ignores them.

We *could* try to process placeholders again to catch those but it makes using shortcodes in reSt in particular very difficult, because almost anything you put in a shortcode is going to be processed by docutils too. For example, if you added a shortcode like this in a reSt file:

```
{{% media http://some.youtube.video %}}
```

By the time we see it it will be something like this:

```
{{% media <a href="http://some.youtube.video">http://some.youtube.video</a> %}}
```

And then it just doesn't work :-(  

BTW: that is why we do that whole "replace with placeholders" dance, docutils breaks most shortcodes all the time.

So, in short: don't use shortcodes in included files. If you have any idea how to do this better I am all ears, but in the meantime, I am closing.

**Comment by RobinD42** [comment]
Thanks for the explanation.
